pkg/web/metricsHandler_test.go

```diff
@@ -0,0 +1,141 @@
+package web
+
+import (
+	"testing"
+
+	"encoding/json"
+	"io/ioutil"
+	"net/http"
+	"net/http/httptest"
+	"github.com/aerogear/aerogear-metrics-api/pkg/mobile"
+	"bytes"
+	"reflect"
+)
+
+type StubMetricsService struct {
+	recorded            mobile.Metric
+	createMethodInvoked bool
+}
+
+func (sms *StubMetricsService) Create(metric mobile.Metric) (mobile.Metric, error) {
+	sms.recorded = metric
+	sms.createMethodInvoked = true
+	return metric, nil
+}
+
```

Comment:
@david-martin @wtrocki here, we created a mock service. Any recommendations for doing it another way? In Java land I would go for Mockito. Here the most starred mocking library I found is https://github.com/golang/mock
Any experience with it?

I will use that one or another one after your feedback.

Reply:
ping @maleck13 for feedback

Reply:
Writing your own mocks is fine. However why not give that lib a go, would be interested to know how useful / time saving it was

Reply:
+1 for creating mock structure as currently implemented in this PR.
No change needed. This is simple api and it should be ok to go this way.
This is fairly common in golang.

Reply:
I have done some reading about this. There are basically 2 ways: monkey patching and mocking.
Monkey patching is possible in Golang as methods can be assigned during run time. I don't think this is a good idea.
Mocking is what we've done. Lots of people do it manually like us, but there are really popular projects on Github that makes mocking easier. I will try one of those projects like GoMock just for learning.

Reply:
@maleck13 @wtrocki 
Instead of Gomock, I went with Testify as it doesn't need any compile-time instrumentation: https://github.com/stretchr/testify 

Commit a7011ca : use Testify for mocking
Commit f915074 : added a missing test case
Commit 039515d : use Testify's assertions

I have some questions that I will post inline though.